### PR TITLE
demo/neural-network-demo

### PR DESCRIPTION
**Pull request by jszuminski**
Added a neural network demo for the demo pages.

---
## Demo
Link to a JSFiddle: https://jsfiddle.net/BlackLabel/3whL2sfe/

## Things to consider
We might consider (in the future or now in this demo) adding a panel of inputs where the user could choose the number of layers and the number of nodes for each of the layers (but wouldn't this be an overkill in a simplistic demo that we want to release now?).

### CODE REVIEW

File: samples/highcharts/demo/neural-network/demo.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    },
+    title: {
+        text: 'Visualizing a neural network with Highcharts'
+    },
+    subtitle: {
+        text: 'You can use the parallel-coordinates module ' +
+            'to visualize a neural network.'
+    },
+    tooltip: {
+        formatter() {
```
Comment (pawelfus) [inline samples/highcharts/demo/neural-network/demo.js:55]: Could you replace formatter with `format`? You can store extra info per series/point/axis.

File: samples/highcharts/demo/neural-network/demo.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+const layers = [1, 6, 6, 6, 2];
```
Comment (pawelfus) [inline samples/highcharts/demo/neural-network/demo.js:1]: Add a comment that those variables can be modified to render different graphs.
Unless we want to go with a fixed dataset, leaving the decision up to @TorsteinHonsi

**Review by pawelfus — CHANGES_REQUESTED** [review]
Looks good!

Could you check why we see chart skew in mobile view? See: 
image
 
I guess this might be related to computed yAxis extremes.- at least I would start there. Alternatively, can we reduce labels to short names (ReLU, tanh etc)?

File: samples/highcharts/demo/neural-network/demo.js (hunk 2 above)
Comment (TorsteinHonsi) [reply samples/highcharts/demo/neural-network/demo.js:1]: +1 for the comment. I'd like to see some more comments in general for this demo.

**Comment by jszuminski** [comment]
I've added the requested changes. This is an updated demo in JSFiddle: https://jsfiddle.net/47uaj3cs/

Also, I was thinking that maybe instead of `layers` and `activation` arrays of numbers, it would be more intuitive for the users to create an array of objects like this:
```
const layers = [{
  nodes: 6,
  activation: 'tanh'
}, {
  nodes: 6,
  activation: 'ReLU'
}, {
  nodes: 6,
  activation: 'ReLU'
}, {
  nodes: 2,
  activation: 'sigmoid'
}]
```

What do you think @pawelfus @TorsteinHonsi?

**Comment by pawelfus** [comment]
Sounds good!

**Comment by jszuminski** [comment]
Here's an updated JSFiddle: https://jsfiddle.net/4p2bLkhw/

Ready for rereview @pawelfus

**Review by pawelfus — APPROVED** [review]
Thanks!

**Comment by pawelfus** [comment]
Adding Marita for a11y review

**Comment by jszuminski** [comment]
Most up to date demo: https://jsfiddle.net/ewcxdh5b/

File: samples/highcharts/demo/neural-network/demo.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    nodes: 6,
+    activation: 'ReLU',
+    label: 'Hidden Layer #3 (ReLU)'
+}, {
+    nodes: 2,
+    activation: 'sigmoid',
+    label: 'Output Layer (sigmoid)'
+}];
+
+function generateData() {
```
[the remark this answers is not in the document]
Comment (jszuminski) [reply samples/highcharts/demo/neural-network/demo.js:26]: I've tried my best to describe it with a few comments, though not sure if that's fully clear.

**Comment by jszuminski** [comment]
@TorsteinHonsi most up to date demo: https://jsfiddle.net/BlackLabel/2a1y5zhe/

I've removed the `colors` array and added it to `plotOptions.line.color`, but I've also reduced the opacity of the lines.

In my opinion, it looks better now. What do you think?

File: samples/highcharts/demo/neural-network/demo.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (TorsteinHonsi) [reply samples/highcharts/demo/neural-network/demo.js:26]: Looks good 👍

**Review by TorsteinHonsi — APPROVED** [review]
Looks good, thank you!

**Review by marvin19 — APPROVED** [review]

**Comment by jszuminski** [comment]
As discussed with @marvin19, we are removing `point.descriptionFormat` temporarily as there are a few problems related to it independent to this pull request.

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks!

1. The `descriptionFormat` works for me like this
```js
    accessibility: {
        typeDescription: 'Neural network chart',
        point: {
            descriptionFormat:
                'node on {series.xAxis.options.custom.layers.(x).label}'
        }
    },
```
Is that okay or are there other problems?


2. The files of the `vendor` folder are added in this PR. Those should be removed.

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**